Branch-SDK/src/io/branch/referral/ServerRequestInitSession.java

```diff
@@ -200,6 +231,65 @@ public void onPreExecute() {
         } catch (JSONException ignore) {
             
         }
+    }
+    
+    /*
+     * Method to update the install or update state along with the timestamps.
+     * PRS NOTE
+     * The Original install time will have a the very first install time only if the app allows preference back up.
+     * Apps that need to distinguish between a fresh install and re-install need to allow backing up of preferences.
+     * Previous install time stamp always carry the last last known update time. the value will be zero for any fresh install and will be the last update time with successive opens.
+     * Previous install time will have a value less than last update time ever since the app is updated.
+     *
+     *  ------------------------------------------------------------
+     * |   update_state_install    | lut <= fit, fit = oit, put = 0 |
+     *  --------------------------- --------------------------------
+     * |   update_state_reinstall  | oit < fit, put = 0             |
+     *  --------------------------- --------------------------------
+     * |   update_state_update     | lut > fit, put < lut           |
+     *  --------------------------- --------------------------------
+     * |   update_state_no_update  | lut == put                     |
+     *  --------------------------- --------------------------------
+     * @param post Post body for init request which need to be updated
+     * @throws JSONException when there is any exception on adding time stamps or update state
+     */
+    private void updateInstallStateAndTimestamps(JSONObject post) throws JSONException {
+        int installOrUpdateState = STATE_NO_CHANGE;
+        String currAppVersion = systemObserver_.getAppVersion();
+        PackageInfo packageInfo = null;
+        try {
+            packageInfo = context_.getPackageManager().getPackageInfo(context_.getPackageName(), 0);
+        } catch (PackageManager.NameNotFoundException ignored) {
+        }
+        if (PrefHelper.NO_STRING_VALUE.equals(prefHelper_.getAppVersion())) {
+            // Default, just register an install
+            installOrUpdateState = STATE_FRESH_INSTALL;
+            // if no app version is in storage, this must be the first time Branch is here
+            if (packageInfo != null && packageInfo.lastUpdateTime != packageInfo.firstInstallTime) {
```

Comment:
Just to be safe for this time compare and the one below on line 288, can you add a 24 hour buffer for equality? i.e. `abs(packageInfo.lastUpdateTime - packageInfo.firstInstallTime) > 24` I worry about the same issue happening here that happened on iOS where the times vary by a minute or two because of an OS change.